### PR TITLE
Name Obfuscation 2026

### PR DESCRIPTION
**Pull request by AnCopper**
# About the pull request

This PR introduces a name obfuscation system for Marines and Xenomorphs. The goal is to strip away the "gamey" feel of chat logs and visible messages during combat and death. Instead of seeing specific player names, enemies will now see generic descriptors based on the target's gender or caste.

Now, a Xeno will see "a bulky/slender tall host", and a Marine will see "a [caste] alien".

The obfuscation extends to examine procs, standard melee attacks, projectile hits, etc. I also updated the status bar tooltips on mouse-over. The goal is to cover anything that could print out a players real name.

One important thing to note: it is impossible to obfuscate the right-click context menu due to hardcoded BYOND limitations. Real names will still show up there, but I was told by Harry that the context menu is likely getting removed eventually anyway.

# Explain why it's good for the game

You can metagame the hell out of combat just by reading chat. People recognize character names mid-combat "Oh that's the good rav player, focus her" or "that's a marine main, let's fuck with him specifically." The game just hands you information you shouldn't have and people absolutely use this info.

And pure immersion. You only know what your character would actually know. That's how it should work.

# Testing Photographs and Procedure

I've run through the standard combat and death cycles for both sides.

Obviously, this is a massive change that touches a ton of different files and standard visible message calls. It’s going to require a lot of hands-on testing to hunt down every single obscure source or item interaction that might still be printing a real name to chat. Consider this a baseline that covers the most frequent combat and interaction messages, but we’ll definitely need to keep an eye out for edge cases that still leak identities. I recommend test merging this for at least a few weeks before merging it into the codebase.

<details>
<summary>Screenshots & Videos</summary>

This is what marines see for a ravager:
grafik
grafik
grafik
grafik
grafik
(It's impossible to obfuscate the context menu)

This is what xenos will see when they attack a male marine:
grafik

</details>

# Changelog

:cl:
add: Xenomorphs and Marines now see generic "Tall Host" or "Alien" descriptors in chat for death, combat, dismemberment, and examinations instead of real player names.
ui: Hover tooltips for enemy factions now show generic descriptors.
/:cl:

### CODE REVIEW

**Comment by ihatethisengine** [comment]
Is it really worth fighting metagaming at the expense of overall game enjoyment? I think it's a lot more fun to fight people you know instead of some randoms. The damage to RP and immersion from such meta is too minor to justify such a heavy-handed change.

Like what is the real harm? You stay more focused when you see OWL prime ravager? Well, you are supposed to be focused regardless. You will try to cap Joe Dafoe to permanently remove them from the round? Well, you are supposed to cap/perma people anyway. Like the ONLY serious issue I can think of is metagibbing, because gib is limited by the CD, and using it on some PFC just because you know they are good is really metagaming (and this is probably ahelpable). Again, the harm is nothing compared to the damage this PR will do to the enjoyment of the game.

File: code/_globalvars/global_lists.dm [hunk 1]
```diff
@@ -114,6 +114,40 @@ GLOBAL_LIST_INIT_TYPED(resin_mark_meanings, /datum/xeno_mark_define, setup_resin
...
+	XENO_CASTE_PREDALIEN    = "nightmarish",
+	XENO_CASTE_QUEEN        = "royal",
+	XENO_CASTE_RAVAGER      = "razor-sharp",
+	XENO_CASTE_RUNNER       = "fast",
+	XENO_CASTE_SENTINEL     = "venomous",
+	XENO_CASTE_SPITTER      = "acid-dripping",
+	XENO_CASTE_WARRIOR      = "powerful"
+))
+
+// Obfuscate Human Names
```
Comment (Detective-Google) [inline code/_globalvars/global_lists.dm:143]: Change these from gender, women can be "bulky" and men can be "slender", etc. 

Potential alternative to consider is just basing it on the armor they're wearing, just having a small list of randomized fairly nondescript adjectives applied to each human on creation, or even basing it off the muscular/lean/etc body type picker in the character creator.

**Comment by cuberound** [comment]
perhaps link the bulky and such to the bodytype chosen in character setup

**Comment by Red-byte3D** [comment]
Im going to just say a few words and if it gets marked as off topic its fine


I understand the frustration everyone can feel while being meta-targetted, i have been meta-targetted a lot of times myself playing warrior or on my marine as spec or whatever and i have complained about this before, but

While it's really annoying and frustrating i feel like this removes a lot of the fun interactions you can have with people in match, not all recognition is bad.


It's fun having interactions with people where you notice who they are and it reall for me atleast pushes the fact that we are all really just a small community of players,
It's probably LRP yes but its always funny to point to a person you know on the opposing team and mess around with them, my question is, how is this going to necessarily stop any sort of metagaming?

Realistically speaking, this wont stop me in any way from looking at a marine as xeno and noticing they are one of my friends, all it does is make all xenos anonymous (This is fine if its the path we want to go)


Some of the best interactions in this game are people i know attacking me when im human while they are playing xeno and we point at eachother after something funny happens, i feel like this will just remove a lot of that while not really adding anything

It's an easy solution for xeno side because all models look the same, but all marines look unique, how exactly will this stop any sort of meta-targeting? People know what most good players look like in-game, people know what armor CO's are wearing, people know what weapons are useable by what, so why exactly was it limited this way? I feel like all this does in a way is just anonimize people to the point of obscuring social fun rather then help people in a game-mechanic sort of way, if that makes sense?

I dont want to say we dont need this pr, because it definetly gets on my nerves when i get gibbed by queen players for "being good at the game", but i really dont know how this will stop that, people can still see what you look like, and i think all this will make is cut down on a lot of the funnier interactions in the game. I dont want this to be seen as a complaint, i just want this to be considered before / if this gets merged

edt;
follow up idea that came to me a bit later is atleast maybe making so theres an optional anonimty for the XENO side how queen tags work

I.E you can hide your tag for marines (so, you're for example, "Prime Warrior" to marines, but "prime warrior chud" to allies. that way this would allow for anonimity if people would choose to enemies, but still remain recognizible by the allies instead of just having to be xx or whatever. this would be optional obv so you can choose it however)

**Comment by Butlerblock** [comment]
This is something that should be left up to a community vote IMO

**Comment by neeshacark** [comment]
does this also hide playtime maturity like elder/prime/whatever?

**Comment by axiomata7** [comment]
Cut the tall and keep the Host. It looks cleaner.

**Review by Git-Nivrak — CHANGES_REQUESTED** [review]
I don't know if this was made using AI or not but the code is not remotely up to standards, there are tons of copypasted code and extremely expensive checks in code that runs very frequently, DNM until fixed

**Comment by Mary-Marilyn** [comment]
I'll get the code up to standard and attempt it another time.
